reference/tools/user-guide.md

```diff
@@ -6,184 +6,271 @@ aliases: ['/docs/dev/how-to/migrate/from-mysql/','/docs/dev/how-to/migrate/incre
 
-### TiDB backup and restore
+The following are the basics of TiDB Binlog:
 
-Mydumper and Loader -> BR:
+- Input/Output:
+    - Input: TiDB Cluster
+    - Output: MySQL, TiDB, Kafka or incremental backup files
+- Supported TiDB versions: v2.1 or later
+- Kubernetes support: Yes. See [TiDB Binlog Cluster Operations](/tidb-in-kubernetes/maintain/tidb-binlog.md) and [TiDB Binlog Drainer Configurations in Kubernetes](/tidb-in-kubernetes/reference/configuration/tidb-drainer.md) for details.
+
+#### CDC (Beta, under development, ETA May/June 2020 with TiDB 4.0)
+
+CDC (Change Data Capture) is a system that collects changelog for key value pairs in TiKV and outputs to downstream systems in transactional order.
+
+- Input/Output: 
+    - Input: TiDB Cluster
+    - Output: MySQL, TiDB, Kafka or incremental backup files
+- Supported TiDB versions: v4.0
+- Document: WIP
+- Kubernetes support: On the development road map, ETA Q2 2020
+
+## Recommended tools for TiDB versions
+
+### Recommended tools for TiDB 3.0 or earlier
+
+- MySQL full data backup: use Mydumper
+- MySQL full data import to TiDB:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: Use DM
+- MySQL incremental data sync to TiDB: use DM
+- TiDB full data backup: Use Mydumper
+- TiDB full data restore:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: Use DM
+- TiDB incremental backup & restore: use TiDB-Binlog
+
+### Recommended tools for TiDB 3.1
+
+- MySQL full data backup: use Mydumper
+- MySQL full data import to TiDB:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: use DM
+- MySQL incremental data sync to TiDB: use DM
+- TiDB full data backup: use BR
+- TiDB full data restore: use BR
+- TiDB incremental backup & restore: use TiDB-Binlog
+
+### Recommended tools for TiDB 4.0
+
+- MySQL full data backup: use Mydumper
+- MySQL full data import to TiDB:
+    - TB scale: use TiDB Lightning
+    - Sub-TB scale: use DM
+- MySQL incremental data sync to TiDB: use DM
+- TiDB full data backup: use BR
+- TiDB full data restore: use BR
+- TiDB incremental backup & restore: use CDC
+
+## Tools evolution roadmap 
+
+- TiDB Full Data Backup:
+    - Mydumper -> BR
+    - Mydumper -> [dumpling](https://github.com/pingcap/dumpling) (under development, replace Lighting in lightweight scenarios)
+- TiDB Full Data Restore:
+    - Loader -> Lightning -> BR
+- MySQL Data Migration:
+    - Mydumper/Loader + Syncer -> Loader + DM -> Lightning + DM -> DM (in the next step, we will integrate Lightning into DM)
+- TiDB Incremental Data Migration:
+    - TiDB Binlog -> CDC
+
+## Full-path data migration solution for TiDB 3.0, 3.1 and 4.0
+
+TiDB 3.0 is the recommended version and is also the most widely adopted version. In addition, TiDB 3.1 and 4.0 will be released this year. The following sections will cover how to migrate data from MySQL to TiDB, between TiDB clusters, and from TiDB to MySQL for each version, as well as how to back up and restore data.
```

Comment:
Since we already released some 3.1 or 4.0 versions, suggest adding `GA`.
```suggestion
TiDB 3.0 is the recommended version and is also the most widely adopted version. In addition, TiDB 3.1 GA and 4.0 GA will be released this year. The following sections will cover how to migrate data from MySQL to TiDB, between TiDB clusters, and from TiDB to MySQL for each version, as well as how to back up and restore data.
```